src/main/java/seedu/address/ui/TagStatisticsPanel.java

```diff
@@ -0,0 +1,98 @@
+package seedu.address.ui;
+
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import com.google.common.eventbus.Subscribe;
+
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.chart.BarChart;
+import javafx.scene.chart.CategoryAxis;
+import javafx.scene.chart.XYChart;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.events.model.AddressBookChangedEvent;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.person.ReadOnlyPerson;
+import seedu.address.model.tag.Tag;
+
+//@@author Valerieyue
+/**
+ * The birthday statistics panel of the App.
+ */
+public class TagStatisticsPanel extends UiPart<Region> {
+
+    private static final String FXML = "TagStatisticsPanel.fxml";
+
+    @FXML
+    private BarChart<String, Integer> barChart;
+
+    @FXML
+    private CategoryAxis xAxis;
+
+    private final Logger logger = LogsCenter.getLogger(this.getClass());
+    private ObservableList<String> tagNames = FXCollections.observableArrayList();
+
+    public TagStatisticsPanel(ReadOnlyAddressBook readOnlyAddressBook) {
+        super(FXML);
+        setPersonData(readOnlyAddressBook);
+        barChart.setCategoryGap(10);
+        barChart.setTitle("Tags Statistics");
+        xAxis.setLabel("Tags");
+        registerAsAnEventHandler(this);
+    }
+
+    private void setPersonData(ReadOnlyAddressBook readOnlyAddressBook) {
+        XYChart.Series<String, Integer> series = new XYChart.Series<>();
+
+        // Get an array with the tags.
+        tagNames.clear();
+        for (ReadOnlyPerson person : readOnlyAddressBook.getPersonList()) {
```

Comment:
you should be able to use `readOnlyAddressBook.getTagList()`. this returns a list of all the tags in the addressbook, so there's no need to check for duplicates.

Reply:
Thanks for the suggestions!